src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/NullableCompatibilitySwitch.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    internal class NullableCompatibilitySwitch<TValue> : ICompatibilitySwitch where TValue : struct
+    {
+        public NullableCompatibilitySwitch(string name)
+        {
+            Name = name;
+        }
+
+        public bool IsValueSet => Value.HasValue;
```

Comment:
This gives an inconsistent meaning to `IsValueSet`. The `interface` is about what this property means: `<see cref="Value"/> property has been set`, not that `Value` has been set to a non-`null` value. Use the existing `CompatibilitySwitch<TValue>` type and, if necessary, special-case `null` elsewhere.